src/module/DownloadPost/DownloadPost.ts

```diff
@@ -0,0 +1,60 @@
+import Settings from '@/core/Settings/Settings';
+import { ModuleSetting, P, PoweruserModule } from '@/types';
+
+export default class DownloadPost implements PoweruserModule {
+    readonly id = 'DownloadPost';
+    readonly name = 'Posts in hoher Auflösung herunterladen';
+    readonly description = 'Fügt einen Button zu jedem Post hinzu, um den Post in höchster Qualität herunterzuladen.';
+    readonly isDownloadButtonEnabled = Settings.get("DownloadImage.settings.download_image");
+
+    async load() {
+      this.initDownloadButton();
+    }
+
+    initDownloadButton() {
+      if(!this.isDownloadButtonEnabled) return;
+      window.addEventListener('locationChange', () => {
```

Comment:
Ich glaub hier macht es wirklich Sinn, die Event-Handler tiefer in p0weruser zu integrieren, sodass wir nur diese in das pr0gramm skript hooken müssen. Ist aber zu groß für diesen PR.

Der Button ist sichtbar, wenn die Initialisierung von p0weruser durch ist, aktuell braucht das 500ms (Siehe #122). Er funktioniert aber nicht, wenn ein Direktlink aufgerufen wird.

Ich denke die pragmatischte Lösung ist es, hier den Code aus dem EventListener auszulagern und dann auch in der `initDownloadButton()` Methode direkt aufzurufen. Das sollte dann für beide Fälle funktionieren.